File: jimmy/courses/cpp/data_type/naninf_test.cc
#include <iostream>
using namespace std;

void test() {
  double num1 = 0.0 / 0.0;
  double num2 = 1.0 / 0.0;
  double num3 = -1.0 / 0.0;
  
  // output variables
  cout << num1 << endl;
  cout << num2 << endl;
  cout << num3 << endl;
}

int main() {
  test();
  return 0;
}

File: jimmy/courses/java/data_type/NanInfTest.java
class TestCls {
  public static void run() {
    double num1 = 0.0 / 0.0;
    double num2 = 1.0 / 0.0;
    double num3 = -1.0 / 0.0;
    
    System.out.println(num1);
    System.out.println(num2);
    System.out.println(num3);
  }
}

public class NanInfTest {
  public static void main(String[] args) {
    TestCls.run();
  }
}